processing/src/test/java/org/apache/druid/query/planning/DataSourceAnalysisTest.java

```diff
@@ -0,0 +1,476 @@
+    Assert.assertTrue(analysis.isGlobal());
+    Assert.assertFalse(analysis.isQuery());
+    Assert.assertEquals(joinDataSource, analysis.getDataSource());
+    Assert.assertEquals(LOOKUP_LOOKYLOO, analysis.getBaseDataSource());
+    Assert.assertEquals(Optional.empty(), analysis.getBaseTableDataSource());
+    Assert.assertEquals(Optional.empty(), analysis.getBaseQuerySegmentSpec());
+    Assert.assertEquals(
+        ImmutableList.of(
+            new PreJoinableClause("1.", LOOKUP_LOOKYLOO, JoinType.INNER, joinClause("1."))
+        ),
+        analysis.getPreJoinableClauses()
+    );
+  }
+
+  @Test
+  public void testJoinLookupToTable()
+  {
+    final JoinDataSource joinDataSource = join(
+        LOOKUP_LOOKYLOO,
+        TABLE_FOO,
+        "1.",
+        JoinType.INNER
+    );
+
+    final DataSourceAnalysis analysis = DataSourceAnalysis.forDataSource(joinDataSource);
+
+    Assert.assertFalse(analysis.isConcreteBased());
+    Assert.assertFalse(analysis.isConcreteTableBased());
+    Assert.assertFalse(analysis.isGlobal());
+    Assert.assertFalse(analysis.isQuery());
+    Assert.assertEquals(joinDataSource, analysis.getDataSource());
+    Assert.assertEquals(LOOKUP_LOOKYLOO, analysis.getBaseDataSource());
+    Assert.assertEquals(Optional.empty(), analysis.getBaseTableDataSource());
+    Assert.assertEquals(Optional.empty(), analysis.getBaseQuerySegmentSpec());
+    Assert.assertEquals(
+        ImmutableList.of(
+            new PreJoinableClause("1.", TABLE_FOO, JoinType.INNER, joinClause("1."))
+        ),
+        analysis.getPreJoinableClauses()
+    );
+  }
+
+  /**
+   * Generate a datasource that joins on a column named "x" on both sides.
+   */
+  private static JoinDataSource join(
+      final DataSource left,
+      final DataSource right,
+      final String rightPrefix,
+      final JoinType joinType
+  )
+  {
+    return JoinDataSource.create(
+        left,
+        right,
+        rightPrefix,
+        joinClause(rightPrefix).getOriginalExpression(),
+        joinType,
+        ExprMacroTable.nil()
+    );
+  }
+
+  /**
+   * Generate a join clause that joins on a column named "x" on both sides.
+   */
+  private static JoinConditionAnalysis joinClause(
+      final String rightPrefix
+  )
+  {
+    return JoinConditionAnalysis.forExpression(
+        StringUtils.format("x == \"%sx\"", rightPrefix),
+        rightPrefix,
+        ExprMacroTable.nil()
+    );
+  }
+
+  /**
+   * Generate a datasource that does a subquery on another datasource. The specific kind of query doesn't matter
+   * much for the purpose of this test class, so it's always the same.
+   */
+  private static QueryDataSource subquery(final DataSource dataSource)
+  {
+    return new QueryDataSource(
+        GroupByQuery.builder()
+                    .setDataSource(dataSource)
+                    .setInterval(new MultipleIntervalSegmentSpec(Collections.singletonList(Intervals.of("2000/3000"))))
```

Comment:
Perhaps create a named constant for the interval since it's in the asserts for several of the new tests

Reply:
Sure, sounds reasonable.